flutter/flutter
# add collectGarbage method for SliverChildDelegate to track which children can be garbage collected

## Description

flutter has [ResizeImage] to finer control of the size of the image in [ImageCache] and is
generally used to reduce the memory footprint of [ImageCache], but it's not enough. In fact, at the most time, we can't get image size(width/height) before image is aleady, so we can't set cache width and cache height in the moment.

it should provide an opportunity to clear image cache when children can be garbage collected. user  will get indexes from [collectGarbage] method and clear image cache by using [ImageProvider.evict()].

## Related Issues

https://github.com/flutter/flutter/issues/19558

## Tests

I added the following tests:

* ‘collectGarbage has correct indices’

* prepare ListView(itemExtent: 100.0,cacheExtent: 0.0,) with TestSliverChildListDelegate which will add log into gcLog when collectGarbage is called.
* drag ListView -600.0 and expected log will be 'collectGarbage indexes=[0, 1, 2, 3, 4, 5]'
* drag ListView 100.0 and expected log will be 'collectGarbage indexes=[11]'

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.


## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*
